LEGO1/mxdsfile.cpp

```diff
@@ -1,6 +1,130 @@
 #include "mxdsfile.h"
 
+#include <stdio.h>
+
+#define SI_MAJOR_VERSION 2
+#define SI_MINOR_VERSION 2
+
+#define FCC_OMNI 0x494e4d4f
+#define FCC_MxHd 0x6448784d
+#define FCC_MxOf 0x664f784d
+
+// OFFSET: LEGO1 0x100cc4b0
+MxDSFile::MxDSFile(const char *filename, unsigned long skipReadingChunks)
+{
+  m_filename = filename;
+  m_skipReadingChunks = skipReadingChunks;
+}
+
+// OFFSET: LEGO1 0x100bfed0
+MxDSFile::~MxDSFile()
+{
+  Close();
+}
+
+// OFFSET: LEGO1 0x100cc590
+long MxDSFile::Open(unsigned long uStyle)
+{
+  // No idea what's stopping this one matching, but I'm pretty
+  // confident it has the correct behavior.
+  memset(&m_io, 0, sizeof(MXIOINFO));
+  if (m_io.Open(m_filename.GetData(), uStyle) != 0) {
+    return -1;
+  }
+
+  m_io.SetBuffer(NULL, 0);
```

Comment:
```suggestion
  m_io.SetBuffer(m_io, 0);
```

Reply:
Not surprising you see difference here. SetBuffer (0x100ccbc0) takes _three_ args, not two. Probably it's inherited from DOS days, where the third arg was used for... something. See its `ret` @0x100CCC05; `ret 0Ch" is purging three stack entries = 3 args.

Reply:
You're right. I forgot about the `this` argument.

Reply:
Made a PR for that earlier: https://github.com/stravant/isle/pull/1

Reply:
Sounds good, merged.

Reply:
For reference, I'm not exactly sure what the return types on the MXIOINFO functions are.

I thought they would exactly match the return types from the Windows mmio API, however there's at least one which clearly does not from the generated assembly, so they seem to be wrappers with slightly different types?

I'll figure it out when I actually decomp them but they don't seem like very important functions to tackle until we're closer to loading.